nshlib/nsh_console.h

```diff
@@ -151,6 +151,12 @@ struct nsh_vtbl_s
 
   bool isctty;
 
+  /* Trace line buffer */
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_DUMP
+  char traceline[CONFIG_NSH_LINELEN];
```

Comment:
but traceline is used only when launching program, isn't good to keep the buffer in the whole nsh life cycle